x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -360,34 +373,31 @@ export class AlertsClient<
 
     const alertsToIndex = [...activeAlertsToIndex, ...recoveredAlertsToIndex];
     if (alertsToIndex.length > 0) {
+      const bulkBody = flatMap(
+        [...activeAlertsToIndex, ...recoveredAlertsToIndex].map((alert: Alert & AlertData) => [
+          getBulkMeta(
+            alert.kibana.alert.uuid,
+            this.fetchedAlerts.indices[alert.kibana.alert.uuid],
+            this.fetchedAlerts.seqNo[alert.kibana.alert.uuid],
+            this.fetchedAlerts.primaryTerm[alert.kibana.alert.uuid],
+            this.isUsingDataStreams()
+          ),
+          alert,
+        ])
+      );
+
       try {
         const response = await esClient.bulk({
           refresh: 'wait_for',
           index: this.indexTemplateAndPattern.alias,
-          require_alias: true,
-          body: flatMap(
-            [...activeAlertsToIndex, ...recoveredAlertsToIndex].map((alert: Alert & AlertData) => [
-              {
-                index: {
-                  _id: alert.kibana.alert.uuid,
-                  // If we know the concrete index for this alert, specify it
-                  ...(this.fetchedAlerts.indices[alert.kibana.alert.uuid]
-                    ? {
-                        _index: this.fetchedAlerts.indices[alert.kibana.alert.uuid],
-                        require_alias: false,
-                      }
-                    : {}),
-                },
-              },
-              alert,
-            ])
-          ),
+          require_alias: !this.isUsingDataStreams(),
+          body: bulkBody,
         });
 
         // If there were individual indexing errors, they will be returned in the success response
         if (response && response.errors) {
           const errorsInResponse = (response.items ?? [])
-            .map((item) => (item && item.index && item.index.error ? item.index.error : null))
+            .map((item) => item?.index?.error || item?.create?.error)
```

Comment:
simplification of existing code, and we need to check for `create` errors as well